pilot/pkg/bootstrap/server.go

```diff
@@ -1348,3 +1355,16 @@ func (s *Server) initReadinessProbes() {
 		s.addReadinessProbe(name, probe)
 	}
 }
+
+func (s *Server) StreamServerOverloadInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	current := s.connectionCounter.Add(1)
+	connectionTotal.RecordInt(current)
+	defer func() {
+		connectionTotal.RecordInt(s.connectionCounter.Add(-1))
+	}()
+	if features.ConnectionLimit > 0 && int64(features.ConnectionLimit) < current {
```

Comment:
use that DiscoveryServer.adsClientCount

Reply:
`DiscoveryServer.adsClients` change in stream handler. It is not synchronized with grpc's dial. I think this will cause some problems.

Reply:
Do we need such precise limits? The grpc's dial time should be very short, and `PILOT_MAX_REQUESTS_PER_SECOND` can reject burst connections. This PR is just to reject the continued increase in connections, right?

Reply:
> This PR is just to limit the continued increase in connections, right?

Yes. There is a gap between qps and connection limit. If use DiscoveryServer.adsClients, this feature will fail in extreme cases. I feel it's better to be more precise.